### PR TITLE
Remove defaulting to devtool: source-map when in production with no devtool enabled

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary

This was causing slow builds for large projects.

Closes #

### Deployment Link

https://terra-dev-site-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Remove defaulting to devtool: source-map when in production with no devtool enabled. This was causing slow builds for large projects.
```
Comment (emilyrohrbough) [inline CHANGELOG.md:6]: I'm wondering if we should add docs on this in the chance someone is using this. I'm fine with the change overall.

**Review by emilyrohrbough — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mjhenkes) [reply CHANGELOG.md:6]: Pretty sure only we used these to instrument our terra-ui site, we could just add the dev-tool there.

**Comment by nramamurth** [comment]
What does this change mean with [TerserPlugin still having sourceMap flag set to true](https://github.com/cerner/terra-toolkit-boneyard/blob/v6.9.0/config/webpack/webpack.config.js#L192) in the base config? Does that affect anything at all?

I guess it wouldn't if `devtool` is not set to source-map, inline-source-map, hidden-source-map and nosources-source-map [1] but I am just curious. 

[1] https://webpack.js.org/plugins/terser-webpack-plugin/#sourcemap

**Review by tbiethman — APPROVED** [review]

**Comment by mjhenkes** [comment]
@nramamurth as i understand it terser would do nothing with the devtool set to false.

**Review by yuderekyu — APPROVED** [review]

**Pull request merged**